# Natural Language Processing
## Professor Eric Shah

Student Name: Mohamed Umar Ranginwala

Course: M.Sc. (Artificial Intelligence & Machine Learning) - Defense Specific – III

Roll Number:

## Q. Perform all preprocessing task on a given dataset and transform it into clean data (Practical)

In [2]:
#Installing NLTK library

In [3]:
pip install nltk

In [4]:
import nltk
nltk.download()

NLTK Downloader
---------------------------------------------------------------------------
    d) Download   l) List    u) Update   c) Config   h) Help   q) Quit
---------------------------------------------------------------------------
Downloader> d

Download which package (l=list; x=cancel)?
  Identifier> l
Packages:
  [ ] abc................. Australian Broadcasting Commission 2006
  [ ] alpino.............. Alpino Dutch Treebank
  [ ] averaged_perceptron_tagger Averaged Perceptron Tagger
  [ ] averaged_perceptron_tagger_ru Averaged Perceptron Tagger (Russian)
  [ ] basque_grammars..... Grammars for Basque
  [ ] bcp47............... BCP-47 Language Tags
  [ ] biocreative_ppi..... BioCreAtIvE (Critical Assessment of Information
                           Extraction Systems in Biology)
  [ ] bllip_wsj_no_aux.... BLLIP Parser: WSJ Model
  [ ] book_grammars....... Grammars from NLTK Book
  [ ] brown............... Brown Corpus
  [ ] brown_tei........... Brown Corpus (TEI XML Version)


True

In [72]:
import numpy as np
import pandas as pd
import re
import nltk
import string

In [73]:
data = pd.read_csv('/content/preprocessing data.csv')

In [74]:
data.head()

id  label                                              tweet
0   1      0  #fingerprint #Pregnancy Test https://goo.gl/h1...
1   2      0  Finally a transparant silicon case ^^ Thanks t...
2   3      0  We love this! Would you go? #talk #makememorie...
3   4      0  I'm wired I know I'm George I was made that wa...
4   5      1  What amazing service! Apple won't even talk to...

In [75]:
data.describe()

id        label
count  7920.000000  7920.000000
mean   3960.500000     0.255808
std    2286.451399     0.436342
min       1.000000     0.000000
25%    1980.750000     0.000000
50%    3960.500000     0.000000
75%    5940.250000     1.000000
max    7920.000000     1.000000

In [76]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7920 entries, 0 to 7919
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   id      7920 non-null   int64 
 1   label   7920 non-null   int64 
 2   tweet   7920 non-null   object
dtypes: int64(2), object(1)
memory usage: 185.8+ KB


In [77]:
data.isnull().sum()

id       0
label    0
tweet    0
dtype: int64

##### There are no null values in the dataset

###Converting the tweets into lower case

In [78]:
data["tweet_lower"] = data["tweet"].str.lower()
data.head()

id  label                                              tweet  \
0   1      0  #fingerprint #Pregnancy Test https://goo.gl/h1...   
1   2      0  Finally a transparant silicon case ^^ Thanks t...   
2   3      0  We love this! Would you go? #talk #makememorie...   
3   4      0  I'm wired I know I'm George I was made that wa...   
4   5      1  What amazing service! Apple won't even talk to...   

                                         tweet_lower  
0  #fingerprint #pregnancy test https://goo.gl/h1...  
1  finally a transparant silicon case ^^ thanks t...  
2  we love this! would you go? #talk #makememorie...  
3  i'm wired i know i'm george i was made that wa...  
4  what amazing service! apple won't even talk to...

###Removing Punctuations from the text
One of the common techniques in text preprocessing is the removal of punctuation from textual data. This process standardizes the text and treats words like 'hurray' and 'hurray!' equally.

It's important to thoughtfully select which punctuation marks to exclude, depending on the specific use case. For example, the string.punctuation module in Python encompasses various punctuation symbols:

!"#$%&'()*+,-./:;<=>?@[\]^_{|}~

You have the flexibility to modify this list of punctuation marks according to your requirements. Let's illustrate this technique with an example from a product review tweet dataset:

Example
Happiness in a box!!! Yey!!! #me #phone #sony #xperia #sonyxperiaZ #new #blessed… http://instagram.com/p/exAJ7tloSM/


By removing punctuation, the tweet becomes:

Happiness in a box Yey me phone sony xperia sonyxperiaZ new blessed httpinstagramcompexAJ7tloSM

In [79]:
print(string.punctuation)

!"#$%&'()*+,-./:;<=>?@[\]^_`{|}~


In [80]:
data.head(5)

id  label                                              tweet  \
0   1      0  #fingerprint #Pregnancy Test https://goo.gl/h1...   
1   2      0  Finally a transparant silicon case ^^ Thanks t...   
2   3      0  We love this! Would you go? #talk #makememorie...   
3   4      0  I'm wired I know I'm George I was made that wa...   
4   5      1  What amazing service! Apple won't even talk to...   

                                         tweet_lower  
0  #fingerprint #pregnancy test https://goo.gl/h1...  
1  finally a transparant silicon case ^^ thanks t...  
2  we love this! would you go? #talk #makememorie...  
3  i'm wired i know i'm george i was made that wa...  
4  what amazing service! apple won't even talk to...

In [81]:
punctuation_to_remove = string.punctuation

In [82]:
def remove_punctuation(text):
    return text.translate(str.maketrans('', '', punctuation_to_remove))

data["tweet_without_punctuation"] = data["tweet_lower"].apply(lambda text: remove_punctuation(text))
data.head(5)

id  label                                              tweet  \
0   1      0  #fingerprint #Pregnancy Test https://goo.gl/h1...   
1   2      0  Finally a transparant silicon case ^^ Thanks t...   
2   3      0  We love this! Would you go? #talk #makememorie...   
3   4      0  I'm wired I know I'm George I was made that wa...   
4   5      1  What amazing service! Apple won't even talk to...   

                                         tweet_lower  \
0  #fingerprint #pregnancy test https://goo.gl/h1...   
1  finally a transparant silicon case ^^ thanks t...   
2  we love this! would you go? #talk #makememorie...   
3  i'm wired i know i'm george i was made that wa...   
4  what amazing service! apple won't even talk to...   

                           tweet_without_punctuation  
0  fingerprint pregnancy test httpsgooglh1mfqv an...  
1  finally a transparant silicon case  thanks to ...  
2  we love this would you go talk makememories un...  
3  im wired i know im george i was made that way ...  
4  what amazing service apple wont even talk to m...

###Removal of stopwords

Stopwords are those common words we use all the time, like 'the' or 'a'. Usually, we skip them when working with text because they don't add much useful information for later analysis. Stopwords are words that don't contribute much to the overall meaning of a text. In English, these could include words like "this," "is," "the," "it," and "so." These words often appear frequently across texts, and when analyzing text, they can sometimes overshadow more important words that carry significant meaning.

Here we are deadling with a dataset of product review tweets, if we have a tweet: "This phone is the best, I love it so much!"



In the example tweet, if we remove stopwords, we might end up with: "phone best, love much!" The words "this," "is," "the," and "it" have been removed because they are stopwords. However, the remaining words like "phone," "best," "love," and "much" convey the key sentiment and information of the tweet.

In [83]:
nltk.download('stopwords')

from nltk.corpus import stopwords
print(stopwords.words('english'))


['i', 'me', 'my', 'myself', 'we', 'our', 'ours', 'ourselves', 'you', "you're", "you've", "you'll", "you'd", 'your', 'yours', 'yourself', 'yourselves', 'he', 'him', 'his', 'himself', 'she', "she's", 'her', 'hers', 'herself', 'it', "it's", 'its', 'itself', 'they', 'them', 'their', 'theirs', 'themselves', 'what', 'which', 'who', 'whom', 'this', 'that', "that'll", 'these', 'those', 'am', 'is', 'are', 'was', 'were', 'be', 'been', 'being', 'have', 'has', 'had', 'having', 'do', 'does', 'did', 'doing', 'a', 'an', 'the', 'and', 'but', 'if', 'or', 'because', 'as', 'until', 'while', 'of', 'at', 'by', 'for', 'with', 'about', 'against', 'between', 'into', 'through', 'during', 'before', 'after', 'above', 'below', 'to', 'from', 'up', 'down', 'in', 'out', 'on', 'off', 'over', 'under', 'again', 'further', 'then', 'once', 'here', 'there', 'when', 'where', 'why', 'how', 'all', 'any', 'both', 'each', 'few', 'more', 'most', 'other', 'some', 'such', 'no', 'nor', 'not', 'only', 'own', 'same', 'so', 'than', '

[nltk_data] Downloading package stopwords to /root/nltk_data...
[nltk_data]   Package stopwords is already up-to-date!


In [84]:
# Define a function to remove stopwords
STOPWORDS = set(stopwords.words('english'))
def remove_stopwords(text):
    """Custom function to remove stopwords"""
    tokens = text.split()
    filtered_tokens = [word for word in tokens if word.lower() not in STOPWORDS]
    return ' '.join(filtered_tokens)

In [85]:
data["tweet_without_stop_words"] = data["tweet_without_punctuation"].apply(remove_stopwords)

In [86]:
data.head()

id  label                                              tweet  \
0   1      0  #fingerprint #Pregnancy Test https://goo.gl/h1...   
1   2      0  Finally a transparant silicon case ^^ Thanks t...   
2   3      0  We love this! Would you go? #talk #makememorie...   
3   4      0  I'm wired I know I'm George I was made that wa...   
4   5      1  What amazing service! Apple won't even talk to...   

                                         tweet_lower  \
0  #fingerprint #pregnancy test https://goo.gl/h1...   
1  finally a transparant silicon case ^^ thanks t...   
2  we love this! would you go? #talk #makememorie...   
3  i'm wired i know i'm george i was made that wa...   
4  what amazing service! apple won't even talk to...   

                           tweet_without_punctuation  \
0  fingerprint pregnancy test httpsgooglh1mfqv an...   
1  finally a transparant silicon case  thanks to ...   
2  we love this would you go talk makememories un...   
3  im wired i know im george i was made that way ...   
4  what amazing service apple wont even talk to m...   

                            tweet_without_stop_words  
0  fingerprint pregnancy test httpsgooglh1mfqv an...  
1  finally transparant silicon case thanks uncle ...  
2  love would go talk makememories unplug relax i...  
3  im wired know im george made way iphone cute d...  
4  amazing service apple wont even talk question ...

### Removal of Frequent words

In the preceding preprocessing phase, we eliminated stopwords by considering linguistic characteristics. However, in scenarios where we possess a corpus specific to a particular domain, there could exist frequently occurring words that hold lesser significance for our analysis.

In [87]:
#Let's identify the most prevalent words and proceed to eliminate them in the subsequent phase.

In [88]:
from collections import Counter
cnt = Counter()
for text in data["tweet_without_stop_words"].values:
    for word in text.split():
        cnt[word] += 1

cnt.most_common(10)

[('iphone', 3676),
 ('apple', 2836),
 ('samsung', 1383),
 ('new', 1137),
 ('phone', 950),
 ('sony', 818),
 ('follow', 719),
 ('…', 703),
 ('ipad', 509),
 ('love', 423)]

In [89]:
all_words = ' '.join(data["tweet_without_stop_words"]).split()
cnt = Counter(all_words)
FREQWORDS = set([w for (w, wc) in cnt.most_common(10)])
FREQWORDS

{'apple',
 'follow',
 'ipad',
 'iphone',
 'love',
 'new',
 'phone',
 'samsung',
 'sony',
 '…'}

In [90]:
def remove_freqwords(text):
    return " ".join([word for word in str(text).split() if word not in FREQWORDS])

In [91]:
data["tweet_without_frequent_words"] = data["tweet_without_stop_words"].apply(remove_freqwords)

In [92]:
data.head()

id  label                                              tweet  \
0   1      0  #fingerprint #Pregnancy Test https://goo.gl/h1...   
1   2      0  Finally a transparant silicon case ^^ Thanks t...   
2   3      0  We love this! Would you go? #talk #makememorie...   
3   4      0  I'm wired I know I'm George I was made that wa...   
4   5      1  What amazing service! Apple won't even talk to...   

                                         tweet_lower  \
0  #fingerprint #pregnancy test https://goo.gl/h1...   
1  finally a transparant silicon case ^^ thanks t...   
2  we love this! would you go? #talk #makememorie...   
3  i'm wired i know i'm george i was made that wa...   
4  what amazing service! apple won't even talk to...   

                           tweet_without_punctuation  \
0  fingerprint pregnancy test httpsgooglh1mfqv an...   
1  finally a transparant silicon case  thanks to ...   
2  we love this would you go talk makememories un...   
3  im wired i know im george i was made that way ...   
4  what amazing service apple wont even talk to m...   

                            tweet_without_stop_words  \
0  fingerprint pregnancy test httpsgooglh1mfqv an...   
1  finally transparant silicon case thanks uncle ...   
2  love would go talk makememories unplug relax i...   
3  im wired know im george made way iphone cute d...   
4  amazing service apple wont even talk question ...   

                        tweet_without_frequent_words  
0  fingerprint pregnancy test httpsgooglh1mfqv an...  
1  finally transparant silicon case thanks uncle ...  
2  would go talk makememories unplug relax smartp...  
3  im wired know im george made way cute daventry...  
4  amazing service wont even talk question unless...

## Removal of Rare words

In [93]:
# Define the number of rare words to keep
num_rare_words = 10

# Create a set of rare words based on frequency
all_words = ' '.join(data["tweet_without_stop_words"]).split()
word_count = Counter(all_words)
rare_words_set = set([word for (word, word_count) in word_count.most_common()[:-num_rare_words-1:-1]])

# Define a function to remove rare words
def remove_rare_words(text):
    """Custom function to remove rare words"""
    words = text.split()
    filtered_words = [word for word in words if word not in rare_words_set]
    return ' '.join(filtered_words)

# Apply the function to the 'tweet_without_stop_words' column and create a new column 'tweet_without_frequent_rare_words'
data["tweet_without_frequent_rare_words"] = data["tweet_without_stop_words"].apply(remove_rare_words)

# Display the updated DataFrame
data.head()

id  label                                              tweet  \
0   1      0  #fingerprint #Pregnancy Test https://goo.gl/h1...   
1   2      0  Finally a transparant silicon case ^^ Thanks t...   
2   3      0  We love this! Would you go? #talk #makememorie...   
3   4      0  I'm wired I know I'm George I was made that wa...   
4   5      1  What amazing service! Apple won't even talk to...   

                                         tweet_lower  \
0  #fingerprint #pregnancy test https://goo.gl/h1...   
1  finally a transparant silicon case ^^ thanks t...   
2  we love this! would you go? #talk #makememorie...   
3  i'm wired i know i'm george i was made that wa...   
4  what amazing service! apple won't even talk to...   

                           tweet_without_punctuation  \
0  fingerprint pregnancy test httpsgooglh1mfqv an...   
1  finally a transparant silicon case  thanks to ...   
2  we love this would you go talk makememories un...   
3  im wired i know im george i was made that way ...   
4  what amazing service apple wont even talk to m...   

                            tweet_without_stop_words  \
0  fingerprint pregnancy test httpsgooglh1mfqv an...   
1  finally transparant silicon case thanks uncle ...   
2  love would go talk makememories unplug relax i...   
3  im wired know im george made way iphone cute d...   
4  amazing service apple wont even talk question ...   

                        tweet_without_frequent_words  \
0  fingerprint pregnancy test httpsgooglh1mfqv an...   
1  finally transparant silicon case thanks uncle ...   
2  would go talk makememories unplug relax smartp...   
3  im wired know im george made way cute daventry...   
4  amazing service wont even talk question unless...   

                   tweet_without_frequent_rare_words  
0  fingerprint pregnancy test httpsgooglh1mfqv an...  
1  finally transparant silicon case thanks uncle ...  
2  love would go talk makememories unplug relax i...  
3  im wired know im george made way iphone cute d...  
4  amazing service apple wont even talk question ...

##URL Removal

The next step in preprocessing entails the removal of any URLs present within the product review dataset.

If we're analyzing product reviews, it's possible that some reviews may include URLs. For the sake of our comprehensive analysis, it might be prudent to exclude these URLs.

In [94]:
# Define a function to remove URLs using str.replace()
def remove_urls(text):
    return re.sub(r'http\S+|www\S+', '', text)

# Apply the function to the 'tweet' column and create a new column 'tweet_without_urls_alt'
data["tweet_without_urls"] = data["tweet_without_frequent_rare_words"].apply(remove_urls)

# Display the updated DataFrame
data.head()

id  label                                              tweet  \
0   1      0  #fingerprint #Pregnancy Test https://goo.gl/h1...   
1   2      0  Finally a transparant silicon case ^^ Thanks t...   
2   3      0  We love this! Would you go? #talk #makememorie...   
3   4      0  I'm wired I know I'm George I was made that wa...   
4   5      1  What amazing service! Apple won't even talk to...   

                                         tweet_lower  \
0  #fingerprint #pregnancy test https://goo.gl/h1...   
1  finally a transparant silicon case ^^ thanks t...   
2  we love this! would you go? #talk #makememorie...   
3  i'm wired i know i'm george i was made that wa...   
4  what amazing service! apple won't even talk to...   

                           tweet_without_punctuation  \
0  fingerprint pregnancy test httpsgooglh1mfqv an...   
1  finally a transparant silicon case  thanks to ...   
2  we love this would you go talk makememories un...   
3  im wired i know im george i was made that way ...   
4  what amazing service apple wont even talk to m...   

                            tweet_without_stop_words  \
0  fingerprint pregnancy test httpsgooglh1mfqv an...   
1  finally transparant silicon case thanks uncle ...   
2  love would go talk makememories unplug relax i...   
3  im wired know im george made way iphone cute d...   
4  amazing service apple wont even talk question ...   

                        tweet_without_frequent_words  \
0  fingerprint pregnancy test httpsgooglh1mfqv an...   
1  finally transparant silicon case thanks uncle ...   
2  would go talk makememories unplug relax smartp...   
3  im wired know im george made way cute daventry...   
4  amazing service wont even talk question unless...   

                   tweet_without_frequent_rare_words  \
0  fingerprint pregnancy test httpsgooglh1mfqv an...   
1  finally transparant silicon case thanks uncle ...   
2  love would go talk makememories unplug relax i...   
3  im wired know im george made way iphone cute d...   
4  amazing service apple wont even talk question ...   

                                  tweet_without_urls  
0  fingerprint pregnancy test  android apps beaut...  
1  finally transparant silicon case thanks uncle ...  
2  love would go talk makememories unplug relax i...  
3  im wired know im george made way iphone cute d...  
4  amazing service apple wont even talk question ...

# Stemming


Stemming involves the process of simplifying inflected or derived words to their fundamental word stem, base, or root form. This procedure aims to achieve word reduction while maintaining the core meaning of words within a given context.

Various stemming algorithms are available, each offering distinct methodologies. One notable approach is the Porter stemmer, widely recognized and embraced within linguistic applications. Leveraging tools like the NLTK package enables the seamless integration of stemming processes for enhanced text analysis.

In [95]:
from nltk.stem.porter import PorterStemmer
# Initialize the PorterStemmer
stemmer = PorterStemmer()

# Define a function to stem words
def stem_words(text):
    return " ".join([stemmer.stem(word) for word in text.split()])

# Apply the function to the 'text' column and create a new column 'text_stemmed'
data["tweet_stemmed"] = data["tweet_without_urls"].apply(stem_words)

# Display the updated DataFrame
data.tail()

id  label                                              tweet  \
7915  7916      0  Live out loud #lol #liveoutloud #selfie #smile...   
7916  7917      0  We would like to wish you an amazing day! Make...   
7917  7918      0  Helping my lovely 90 year old neighbor with he...   
7918  7919      0  Finally got my #smart #pocket #wifi stay conne...   
7919  7920      0  Apple Barcelona!!! #Apple #Store #BCN #Barcelo...   

                                            tweet_lower  \
7915  live out loud #lol #liveoutloud #selfie #smile...   
7916  we would like to wish you an amazing day! make...   
7917  helping my lovely 90 year old neighbor with he...   
7918  finally got my #smart #pocket #wifi stay conne...   
7919  apple barcelona!!! #apple #store #bcn #barcelo...   

                              tweet_without_punctuation  \
7915  live out loud lol liveoutloud selfie smile son...   
7916  we would like to wish you an amazing day make ...   
7917  helping my lovely 90 year old neighbor with he...   
7918  finally got my smart pocket wifi stay connecte...   
7919  apple barcelona apple store bcn barcelona trav...   

                               tweet_without_stop_words  \
7915  live loud lol liveoutloud selfie smile sony mu...   
7916  would like wish amazing day make every minute ...   
7917  helping lovely 90 year old neighbor ipad morni...   
7918  finally got smart pocket wifi stay connected a...   
7919  apple barcelona apple store bcn barcelona trav...   

                           tweet_without_frequent_words  \
7915  live loud lol liveoutloud selfie smile music h...   
7916  would like wish amazing day make every minute ...   
7917  helping lovely 90 year old neighbor morning ma...   
7918  finally got smart pocket wifi stay connected a...   
7919  barcelona store bcn barcelona travel selfie fl...   

                      tweet_without_frequent_rare_words  \
7915   live loud lol selfie smile sony music headphones   
7916  would like wish amazing day make every minute ...   
7917  helping lovely 90 year old ipad morning made r...   
7918  finally got smart pocket wifi stay connected i...   
7919  apple barcelona apple store barcelona travel i...   

                                     tweet_without_urls  \
7915   live loud lol selfie smile sony music headphones   
7916  would like wish amazing day make every minute ...   
7917  helping lovely 90 year old ipad morning made r...   
7918  finally got smart pocket wifi stay connected i...   
7919  apple barcelona apple store barcelona travel i...   

                                          tweet_stemmed  
7915      live loud lol selfi smile soni music headphon  
7916  would like wish amaz day make everi minut coun...  
7917  help love 90 year old ipad morn made realis do...  
7918  final got smart pocket wifi stay connect ipad ...  
7919  appl barcelona appl store barcelona travel iph...

Here in id: 7919 `connected` was changed to `connect`

Consider the following examples extracted from a product reviews dataset:


Original Text: "lost all pictures and everything on my phone from the past year"

Stemmed Version: "lost all pictur and everyth on my phone from the past year"
In the example, "pictures" becomes "pictur." While stemming retains the core meaning, it streamlines the vocabulary for analysis, enhancing text processing tasks.

## Lemmatization
Lemmatization is a text preprocessing technique in natural language processing (NLP) that involves reducing words to their base or root form, known as a "lemma." The goal of lemmatization is to transform different inflected forms of a word into a common base form so that they can be treated as the same word during analysis. This process helps in standardizing and simplifying words while retaining their essential meaning.

Unlike stemming, which involves removing prefixes and suffixes from words to obtain their root form, lemmatization ensures that the resulting lemma is a valid word in the language. This linguistic validation is achieved by using a dictionary or vocabulary (such as WordNet in NLTK) to map words to their lemmatized forms.

For example:

Lemmatization of the word "running" might result in the lemma "run."
Lemmatization of the word "better" might result in the lemma "good."

In [97]:
from nltk.corpus import wordnet
from nltk.stem import WordNetLemmatizer

# Download NLTK WordNet and POS tagger data
nltk.download('wordnet')
nltk.download('averaged_perceptron_tagger')


# Initialize the WordNetLemmatizer
lemmatizer = WordNetLemmatizer()

# Map POS tags to WordNet POS tags
wordnet_map = {"N": wordnet.NOUN, "V": wordnet.VERB, "J": wordnet.ADJ, "R": wordnet.ADV}

# Define a function to lemmatize words with POS tagging
def lemmatize_words(text):
    pos_tagged_text = nltk.pos_tag(text.split())
    return " ".join([lemmatizer.lemmatize(word, wordnet_map.get(pos[0], wordnet.NOUN)) for word, pos in pos_tagged_text])

# Apply the function to the 'tweet' column and create a new column 'text_lemmatized'
data["text_lemmatized"] = data["tweet_without_urls"].apply(lemmatize_words)

# Display the updated DataFrame
data.tail()


[nltk_data] Downloading package wordnet to /root/nltk_data...
[nltk_data]   Package wordnet is already up-to-date!
[nltk_data] Downloading package averaged_perceptron_tagger to
[nltk_data]     /root/nltk_data...
[nltk_data]   Package averaged_perceptron_tagger is already up-to-
[nltk_data]       date!


id  label                                              tweet  \
7915  7916      0  Live out loud #lol #liveoutloud #selfie #smile...   
7916  7917      0  We would like to wish you an amazing day! Make...   
7917  7918      0  Helping my lovely 90 year old neighbor with he...   
7918  7919      0  Finally got my #smart #pocket #wifi stay conne...   
7919  7920      0  Apple Barcelona!!! #Apple #Store #BCN #Barcelo...   

                                            tweet_lower  \
7915  live out loud #lol #liveoutloud #selfie #smile...   
7916  we would like to wish you an amazing day! make...   
7917  helping my lovely 90 year old neighbor with he...   
7918  finally got my #smart #pocket #wifi stay conne...   
7919  apple barcelona!!! #apple #store #bcn #barcelo...   

                              tweet_without_punctuation  \
7915  live out loud lol liveoutloud selfie smile son...   
7916  we would like to wish you an amazing day make ...   
7917  helping my lovely 90 year old neighbor with he...   
7918  finally got my smart pocket wifi stay connecte...   
7919  apple barcelona apple store bcn barcelona trav...   

                               tweet_without_stop_words  \
7915  live loud lol liveoutloud selfie smile sony mu...   
7916  would like wish amazing day make every minute ...   
7917  helping lovely 90 year old neighbor ipad morni...   
7918  finally got smart pocket wifi stay connected a...   
7919  apple barcelona apple store bcn barcelona trav...   

                           tweet_without_frequent_words  \
7915  live loud lol liveoutloud selfie smile music h...   
7916  would like wish amazing day make every minute ...   
7917  helping lovely 90 year old neighbor morning ma...   
7918  finally got smart pocket wifi stay connected a...   
7919  barcelona store bcn barcelona travel selfie fl...   

                      tweet_without_frequent_rare_words  \
7915   live loud lol selfie smile sony music headphones   
7916  would like wish amazing day make every minute ...   
7917  helping lovely 90 year old ipad morning made r...   
7918  finally got smart pocket wifi stay connected i...   
7919  apple barcelona apple store barcelona travel i...   

                                     tweet_without_urls  \
7915   live loud lol selfie smile sony music headphones   
7916  would like wish amazing day make every minute ...   
7917  helping lovely 90 year old ipad morning made r...   
7918  finally got smart pocket wifi stay connected i...   
7919  apple barcelona apple store barcelona travel i...   

                                          tweet_stemmed  \
7915      live loud lol selfi smile soni music headphon   
7916  would like wish amaz day make everi minut coun...   
7917  help love 90 year old ipad morn made realis do...   
7918  final got smart pocket wifi stay connect ipad ...   
7919  appl barcelona appl store barcelona travel iph...   

                                        text_lemmatized  
7915    live loud lol selfie smile sony music headphone  
7916  would like wish amazing day make every minute ...  
7917  help lovely 90 year old ipad morning make real...  
7918  finally get smart pocket wifi stay connect ipa...  
7919  apple barcelona apple store barcelona travel i...

Here in id 7919	we see that `got` was converted to `get` when Lemmetization was applied